crates/primitives-traits/src/account.rs

```diff
@@ -165,19 +165,25 @@ impl reth_codecs::Compact for Bytecode {
 
         use compact_ids::*;
 
-        let len = buf.read_u32::<byteorder::BigEndian>().expect("could not read bytecode length");
-        let bytes = Bytes::from(buf.copy_to_bytes(len as usize));
+        let original_len = buf
+            .read_u32::<byteorder::BigEndian>()
+            .expect("could not read bytecode length") as usize;
+        let bytes = Bytes::from(buf.copy_to_bytes(original_len));
         let variant = buf.read_u8().expect("could not read bytecode variant");
         let decoded = match variant {
             LEGACY_RAW_BYTECODE_ID => Self(RevmBytecode::new_raw(bytes)),
             REMOVED_BYTECODE_ID => {
                 unreachable!("Junk data in database: checked Bytecode variant was removed")
             }
-            LEGACY_ANALYZED_BYTECODE_ID => Self(RevmBytecode::new_analyzed(
-                bytes,
-                buf.read_u64::<byteorder::BigEndian>().unwrap() as usize,
-                revm_bytecode::JumpTable::from_slice(buf),
-            )),
+            LEGACY_ANALYZED_BYTECODE_ID => {
+                let jump_table_len =
+                    if buf.len() * 8 >= bytes.len() { bytes.len() } else { original_len };
```

Comment:
hmm, can we document this here not obvious why we need this or why this is correct

because we're still reading from that buf in L183

Reply:
Added comment